File: leetcode/0128-longest-consecutive-sequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size()==0) return 0;
        sort(nums.begin(),nums.end());
        int maxseq=1;
        int ctr=1;
        for (int i=0;i<nums.size()-1;i++){
            if (nums[i]==nums[i+1]) continue;
            if (nums[i]+1==nums[i+1]) ctr++;
            else ctr=1;
            if (ctr>maxseq){
                maxseq=ctr;
            }
        }
        return maxseq;
    }
};

File: leetcode/0128-longest-consecutive-sequence/solution.py
class Solution(object):
    def longestConsecutive(self, nums):
        nums.sort()
        n=len(nums)
        ctr=1
        maxsize=1
        if(n==0):
            return 0
        for i in range(n-1):
            if(nums[i]==nums[i+1]):
                continue
            elif(nums[i]+1==nums[i+1]):
                ctr+=1
            else:
                ctr=1
            maxsize=max(maxsize,ctr)
            
        return maxsize
        """
        :type nums: List[int]
        :rtype: int
        """
        
